drivers/modem/modem_cmd_handler.c

```diff
@@ -645,28 +645,70 @@ void modem_cmd_handler_tx_unlock(struct modem_cmd_handler *handler)
 	k_sem_give(&data->sem_tx_lock);
 }
 
-int modem_cmd_handler_init(struct modem_cmd_handler *handler,
-			   struct modem_cmd_handler_data *data)
+int modem_cmd_handler_init(struct modem_cmd_handler *handler, struct modem_cmd_handler_data *data,
+			      char *match_buf, size_t match_buf_len, struct net_buf_pool *buf_pool,
+			      k_timeout_t alloc_timeout, const char *eol, void *user_data)
 {
-	if (!handler || !data) {
+	/* Verify arguments */
+	if (!handler || !data || !match_buf || !match_buf_len || !buf_pool) {
 		return -EINVAL;
 	}
 
-	if (!data->match_buf_len) {
-		return -EINVAL;
-	}
+	/* Assign data to command handler */
+	handler->cmd_handler_data = data;
+
+	/* Assign command process implementation to command handler */
+	handler->process = cmd_handler_process;
 
-	if (data->eol == NULL) {
+	/* Store arguments */
+	data->match_buf = match_buf;
+	data->match_buf_len = match_buf_len;
+	data->buf_pool = buf_pool;
+	data->alloc_timeout = alloc_timeout;
+
+	/* Process end of line */
+	if (!eol) {
+		data->eol = NULL;
```

Comment:
looks like you can just set `data->eol` to `eol` outside of this if, and set the len accordingly with this if statement